fl4health/model_bases/masked_layers/masked_normalization_layers.py

```diff
@@ -0,0 +1,297 @@
+from typing import List, Optional, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch import Tensor
+from torch.nn.modules.batchnorm import _BatchNorm
+from torch.nn.parameter import Parameter
+
+from fl4health.utils.functions import bernoulli_sample
+
+_shape_t = Union[int, List[int], torch.Size]
+
+
+class MaskedLayerNorm(nn.LayerNorm):
+    def __init__(
+        self,
+        normalized_shape: _shape_t,
+        eps: float = 1e-5,
+        elementwise_affine: bool = True,
+        bias: bool = True,
+        device: Optional[torch.device] = None,
+        dtype: Optional[torch.dtype] = None,
+    ) -> None:
+        """
+        Implementation of the masked Layer Normalization module. When elementwise_affine is True,
+        nn.LayerNorm has a learnable weight and (optional) bias. For MaskedLayerNorm,
+        the weight and bias do not receive gradient in back propagation.
+        Instead, two score tensors - one for the weight and another for the bias - are maintained.
+        In the forward pass, the score tensors are transformed by the Sigmoid function into probability scores,
+        which are then used to produce binary masks via bernoulli sampling.
+        Finally, the binary masks are applied to the weight and the bias. During training,
+        gradients with respect to the score tensors are computed and used to update the score tensors.
+
+        When elementwise_affine is False, nn.LayerNorm does not have weight or bias.
+        Under this condition, both score tensors are None and MaskedLayerNorm acts in the same way as nn.LayerNorm.
+
+        Note: the scores are not assumed to be bounded between 0 and 1.
+
+        Args:
+            normalized_shape (int or list or torch.Size): input shape from an expected input.
```

Comment:
Super minor, but I'd update the type for `normalized_shape` to be whatever you decide the type alias should called.